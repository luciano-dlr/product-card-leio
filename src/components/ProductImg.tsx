import React from 'react';
import { CSSProperties, useContext } from "react";
import NoImage from "../assets/no-image.jpg";
import styles from "../styles/styles.module.css";
import ProductContext from "./ProductCardContext";


export interface ProductCardImgProps {
  className?: string;
  style?: CSSProperties;
  img?: string;
}

export const ProductImg = ({ className, style, img }: ProductCardImgProps) => {
  const { product } = useContext(ProductContext);
  // const { img } = product;


  let imgToShow: string;

  if (img) {
    imgToShow = img;
  } else if (product.img) {
    imgToShow = product.img
  } else {
    imgToShow = NoImage;
  }

  return (
    <img
      className={`${styles.productImg} ${className}`}
      src={imgToShow}
      style={style}
      alt="Product"
    />
  );
};
