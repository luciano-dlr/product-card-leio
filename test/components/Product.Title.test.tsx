import React from "react"
import renderer from "react-test-renderer"
import { ProductTitle } from '../../src/components/ProductTitle';

describe('Product Title', () => {

    test('Debe de mostrar el componente correctamente con el titulo personalizado ', () => {

        const wraper = renderer.create(
            <ProductTitle title="Custom Product" className="custom-class" />
        )

        expect(wraper.toJSON()).toMatchSnapshot();


    })




})