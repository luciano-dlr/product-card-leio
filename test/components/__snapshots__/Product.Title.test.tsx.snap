// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Title Debe de mostrar el componente correctamente con el titulo personalizado  1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
